### PR TITLE
Dashboard Proposal List and Search Bar

### PR DESCRIPTION
**Pull request by pao-prog**
## Motivation

Created a way to display proposals in a list and search bar to find certain proposals quickly.

Issue #1 

## Summary of Changes

- Added a row component.
- Added a search bar component + functionality.
- Displayed components on the left side of our dashboard.

## Testing

Manual Testing.

## Screenshot/Gif of results
tralalalalalala

### CODE REVIEW

File: client/src/App.js [hunk 1]
```diff
@@ -1,23 +1,25 @@
...
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
+      <Dashboard/>
+      {/* HELLO WORLD
+      <FeatherIcon.X color="#dd515f" size={40}/>
+      <FeatherIcon.Check color="#c9e5cb" size={40}/> */}
```
Comment (Warrenwwang1) [inline client/src/App.js]: remove this if it's unnecessary

File: client/src/App.js [hunk 2]
```diff
@@ -1,23 +1,25 @@
...
+let docWidth = document.documentElement.offsetWidth;
+
+[].forEach.call(
+  document.querySelectorAll('*'),
+  function(el) {
+    if (el.offsetWidth > docWidth) {
+      console.log(el);
+    }
+  }
+);
```
Comment (Warrenwwang1) [inline client/src/App.js]: remove if unnecessary

File: client/package.json [hunk 3]
```diff
@@ -8,8 +8,10 @@
     "@testing-library/user-event": "^12.1.10",
     "react": "^17.0.1",
     "react-dom": "^17.0.1",
+    "react-feather": "^2.0.9",
     "react-scripts": "4.0.3",
-    "web-vitals": "^1.0.1"
+    "web-vitals": "^1.0.1",
+    "yarn": "^1.22.10"
```
Comment (Warrenwwang1) [inline client/package.json]: We don't want yarn here as a dependency here right @KnightAsterial ? what would be the best way to remove

File: package.json [hunk 4]
```diff
@@ -5,5 +5,8 @@
   "main": "index.js",
   "repository": "https://github.com/codebase-berkeley/tgif-voting-portal.git",
   "author": "Warren <warrenwwang@berkeley.edu>",
-  "license": "MIT"
+  "license": "MIT",
+  "dependencies": {
+    "yarn": "^1.22.10"
```
Comment (Warrenwwang1) [inline package.json]: same w yarn here @KnightAsterial

File: client/src/pages/dashboard/Dashboard.js [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+  for (let i = 0; i < proposals.length; i++) {
+    proposalHTML.push(<Row title={proposals[i].title} vote={proposals[i].voted ? proposals[i].voted : ""} />);
+  }
+
+  const [input, setInput] = React.useState("");
+  const [proposalListDefault, setProposalListDefault] = React.useState(proposals);
+  const [proposalList, setProposalList] = React.useState(proposalHTML);
+
+  const updateInput = (input) => {
+    console.log("updated " + input)
```
Comment (Warrenwwang1) [inline client/src/pages/dashboard/Dashboard.js]: remove

File: client/src/pages/dashboard/Dashboard.js [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+import Row from '../../components/row/Row';
+import './Dashboard.css';
+import noIcon from '../../assets/Delete.svg';
+import yesIcon from '../../assets/Checked.svg';
+import SearchBar from '../../components/searchbar/SearchBar';
+
+const proposals = [
+  {
+      title: "some title",
+      voted: yesIcon
```
Comment (Warrenwwang1) [inline client/src/pages/dashboard/Dashboard.js:12]: voted will prob be a boolean from backend fyi

File: client/package.json (hunk 3 above)
Comment (ryanhzhao) [inline client/package.json]: delete this line (remnants of us accidentally running npm commands in a yarn project)

File: client/src/App.js (hunk 2 above)
Comment (ryanhzhao) [inline client/src/App.js]: Remember to delete test code before submitting your pull request to main!

File: client/src/App.js (hunk 1 above)
Comment (ryanhzhao) [inline client/src/App.js]: Same as above, remember to remove comments before PRing changes to main!

File: client/src/components/row/Row.css [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+    font-family: 'Lexend Deca', sans-serif;
+    color: black;
+    font-size: max(2vw, 1.5vw);
+}
+
+.vote-status {
+    display: flex;
+    justify-content: center;
+    width: 3em;
+    height: 3em;
```
Comment (Warrenwwang1) [inline client/src/components/row/Row.css:27]: why are you using em as units for width and height?

File: client/src/components/row/Row.css [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+    flex-direction: row;
+    justify-content: space-between;
+    min-height: 2.5em;
+}
+
+.proposal-title {
+    flex-direction: row;
+    font-family: 'Lexend Deca', sans-serif;
+    color: black;
+    font-size: max(2vw, 1.5vw);
```
Comment (ryanhzhao) [inline client/src/components/row/Row.css]: It's good convention to keep font-size constant in terms of EM rather than dependent on VH

Comment (Warrenwwang1) [inline client/src/components/row/Row.css]: shouldn't font sizes always be in terms of em?

**Review by Warrenwwang1 — CHANGES_REQUESTED** [review]

File: client/src/pages/dashboard/Dashboard.js [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+
+
+function Dashboard() {
+  const proposalHTML = []
+  for (let i = 0; i < proposals.length; i++) {
+    proposalHTML.push(<Row title={proposals[i].title} vote={proposals[i].voted ? proposals[i].voted : ""} />);
+  }
+
+  const [input, setInput] = React.useState("");
+  const [proposalListDefault, setProposalListDefault] = React.useState(proposals);
```
Comment (ryanhzhao) [inline client/src/pages/dashboard/Dashboard.js:63]: is proposalListDefault meant to simply store all the proposals before filtering? If so, you can just use a normal variable for this (it doesn't have to be a state variable)

File: package.json [hunk 10]
```diff
@@ -5,5 +5,8 @@
   "main": "index.js",
   "repository": "https://github.com/codebase-berkeley/tgif-voting-portal.git",
   "author": "Warren <warrenwwang@berkeley.edu>",
-  "license": "MIT"
+  "license": "MIT",
+  "dependencies": {
```
Comment (ryanhzhao) [inline package.json]: likewise with above comment on package.json, can you delete the "dependencies" and "yarn" (lines 9-11)

**Review by ryanhzhao — CHANGES_REQUESTED** [review]
Great code! Super good job with the search bar and making the row components

File: client/src/pages/dashboard/Dashboard.js [hunk 11]
```diff
@@ -0,0 +1,92 @@
...
+      setProposalList(filteredList);
+    } 
+  }
+
+  return (
+    <div className="dashboard">
+      <div className="navBar">this is the delicious navbar</div>
+      <div className="dashboard-screen">
+        <div className="proposal-list">
+          {<SearchBar keyword={input} setKeyword={updateInput}/>}
```
Comment (Warrenwwang1) [inline client/src/pages/dashboard/Dashboard.js]: remove curly bracket

**Review by Warrenwwang1 — APPROVED** [review]
Good job!

**Review by ryanhzhao — APPROVED** [review]

**Pull request merged**